functions/write_file.R

```diff
@@ -10,37 +10,89 @@
 ################################################################################
 
 write_file <- function(data, path, ...) {
-  valid_extensions <- c("rds", "csv")
-  
-  ext <- fs::path_ext(path)
-  
-  if (!(ext %in% valid_extensions)) {
+  if(file.size(path) == 0){
+    valid_extensions <- c("rds", "csv")
+    
+    ext <- fs::path_ext(path)
+    
+    if (!(ext %in% valid_extensions)) {
+      cli::cli_abort(c(
+        "x" = "Invalid extension: {.val {ext}}",
+        "i" = "{.fun read_file} supports {.val {valid_extensions}}"
+      ))
+    }
+    
+    switch(ext,
+           "rds" = readr::write_rds(
+             x = data,
+             file = path,
+             compress = "xz",
+             version = 3L,
+             ...,
+             compression = 9L
+           ),
+           "csv" = readr::write_csv(
+             x = data,
+             file = path,
+             ...
+           )
+    )
+    
+    if (fs::file_info(path)$user == Sys.getenv("USER")) {
+      # Set the correct permissions
+      fs::file_chmod(path = path, mode = "770")
+    }
+    
+    return(invisible(data))
+  }
```

Comment:
I'm finding this part hard to understand because the lines are being repeated further down. Is there a way to simplify this further?

Reply:
aww that makes sense! I'll have a think but there could be a way of not repeating the code. I had forgot about that instance where if we are creating it for the first time we still want to write it first!